src/dkg/round2.rs

```diff
@@ -118,9 +127,66 @@ pub fn import_secret_package(
     SerializableSecretPackage::deserialize_from(&serialized[..]).map(|pkg| pkg.into())
 }
 
+pub fn round2<'a, P, R: RngCore + CryptoRng>(
+    self_identity: &participant::Identity,
+    round1_secret_package: round1::SecretPackage,
+    round1_public_packages: P,
+    mut csrng: R,
+) -> Result<(Vec<u8>, BTreeMap<Identifier, Package>), Error>
+where
+    P: IntoIterator<Item = &'a PublicPackage>,
+    R: RngCore + CryptoRng,
+{
+    let mut round1_public_packages = round1_public_packages.into_iter().collect::<Vec<_>>();
+    round1_public_packages.sort_unstable();
+    round1_public_packages.dedup();
+    let round1_public_packages = round1_public_packages;
+
+    let self_public_package = *round1_public_packages
+        .iter()
+        .find(|&p| p.identity() == self_identity)
+        .expect("missing public package for self_identity");
+
+    let mut frost_packages: BTreeMap<Identifier, round1::Package> = BTreeMap::new();
+    let mut group_secret_key_shards: Vec<&GroupSecretKeyShard> = Vec::new();
+
+    for public_package in round1_public_packages {
+        if public_package.checksum() != self_public_package.checksum() {
+            return Err(Error::ChecksumError(ChecksumError));
```

Comment:
Can we provide a more error message? I don't remember how we do it for TKD (if the error message is generated on ironfish-frost, ironfish-rust, ironfish-rust-nodejs, or typescript), but ideally we should use the same pattern.

This doesn't need to be done in this PR though.

Reply:
I'll update this in a separate PR since it will affect signing_commitment too